SignalServiceKit/src/Storage/OWSStorage.m

```diff
@@ -450,9 +461,7 @@ - (void)resetStorage
 
 + (void)resetAllStorage
 {
-    for (OWSStorage *storage in self.allStorages) {
-        [storage resetStorage];
-    }
+    [[NSNotificationCenter defaultCenter] postNotificationName:OWSResetStorageNotification object:nil];
```

Comment:
Can you help me understand this change - what was wrong with the old approach?

Reply:
We reset storage in a number of scenarios including failure to load key from keychain.  This happens while initializing `TSStorageManager`. Therefore we can't iterate over allStorages 
 - by definition those storages might not be initialized or might be in the process of initializing.  

Correct approach - fire a notification so that any pre-existing storage instances reset.